core-api/src/main/java/com/optimizely/ab/config/audience/MatchType.java

```diff
@@ -0,0 +1,187 @@
+package com.optimizely.ab.config.audience;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+interface LeafMatcher {
+    public Boolean eval(Object otherValue);
+}
+
+abstract class LeafMatch<T> implements LeafMatcher {
+    T convert(Object o) {
```

Comment:
Please document this method.

Reply:
Also, I think we should be letting these throw so we can log the type mismatch as a debug level log?